server/tserver/src/main/java/org/apache/accumulo/tserver/InMemoryMap.java

```diff
@@ -683,13 +683,6 @@ public void setInterruptFlag(AtomicBoolean flag) {
     private SourceSwitchingIterator ssi;
     private MemoryDataSource mds;
 
-    @Override
-    protected SortedKeyValueIterator<Key,Value> getSource() {
-      if (closed.get())
-        throw new IllegalStateException("Memory iterator is closed");
-      return super.getSource();
-    }
-
```

Comment:
I did a bit of research into the implications of removing this sanity check.  The case where the check would have triggered is when scan iterators are used after Accumulo has finished using the iterator stack.  If that's what we want to check for, this is probably not the best place to do it.